tests/MongoDB.Bson.Tests/Specifications/bson-corpus/BsonCorpusTestRunner.cs

```diff
@@ -0,0 +1,358 @@
+            }
+        }
+
+        // private methods
+        private void AssertException(Exception exception)
+        {
+            exception
+                .Should()
+                .Match<Exception>(
+                    e =>
+                        e is FormatException ||
+                        e is IOException ||
+                        e is ArgumentException ||
+                        e is IndexOutOfRangeException ||
+                        e is OverflowException);
+        }
+
+        private BsonDocument DecodeBson(byte[] bytes)
+        {
+#pragma warning disable 618
+            var readerSettings = new BsonBinaryReaderSettings
+            {
+                FixOldBinarySubTypeOnInput = false,
+                FixOldDateTimeMaxValueOnInput = false
+            };
+            if (BsonDefaults.GuidRepresentationMode == GuidRepresentationMode.V2)
+            {
+                readerSettings.GuidRepresentation = GuidRepresentation.Unspecified;
+            }
+#pragma warning restore 618
+            using (var stream = new MemoryStream(bytes))
+            using (var reader = new BsonBinaryReader(stream, readerSettings))
+            {
+                var context = BsonDeserializationContext.CreateRoot(reader);
+                return BsonDocumentSerializer.Instance.Deserialize(context);
+            }
+        }
+
+        private BsonDocument DecodeExtendedJson(string extendedJson)
+        {
+            return BsonDocument.Parse(extendedJson);
+        }
+
+        private byte[] EncodeBson(BsonDocument document)
+        {
+#pragma warning disable 618
+            var writerSettings = new BsonBinaryWriterSettings
+            {
+                FixOldBinarySubTypeOnOutput = false,
+            };
+            if (BsonDefaults.GuidRepresentationMode == GuidRepresentationMode.V2)
+            {
+                writerSettings.GuidRepresentation = GuidRepresentation.Unspecified;
+            }
+#pragma warning restore 618
+            using (var stream = new MemoryStream())
+            using (var writer = new BsonBinaryWriter(stream, writerSettings))
+            {
+                var context = BsonSerializationContext.CreateRoot(writer);
+                BsonDocumentSerializer.Instance.Serialize(context, document);
+                return stream.ToArray();
+            }
+        }
+
+        private string EncodeCanonicalExtendedJson(BsonDocument document)
+        {
+#pragma warning disable 618
+            var writerSettings = new JsonWriterSettings
+            {
+                OutputMode = JsonOutputMode.CanonicalExtendedJson,
+            };
+            if (BsonDefaults.GuidRepresentationMode == GuidRepresentationMode.V2)
+            {
+                writerSettings.GuidRepresentation = GuidRepresentation.Unspecified;
+            }
+#pragma warning restore 618
+            var json = document.ToJson(writerSettings);
+            return json.Replace(" ", "");
```

Comment:
Removing all blanks seems dangerous, and only works because none of the test strings have blanks in them.

However, I supposed it's the easiest course of action for now.